WordPress/gutenberg
# Improve cover z-index solution

## What?


This is an improved solution for this original PR: https://github.com/WordPress/gutenberg/pull/66093

As part of this PR, I also thought of removing [this effect that adds the `has-modal-open` in the editor](https://github.com/WordPress/gutenberg/pull/66093/files#diff-884dae78f1676d454c049107e9b05b2403eb042457f77f836bd170d104396b8bR89-R97) because that wouldn't be needed for this purpose anymore. But thinking more, I kept it to keep the consistency there between the frontend and the editor.

## Why?


To improve the solution and eliminate the dependencies between Cover and Navigation blocks (the class that Navigation adds being used in the Cover styles).

More context starting from [this comment](https://github.com/WordPress/gutenberg/pull/66093/files#r1804730040) in the thread.

## How?


It deprecates the current version of the Cover block, changing the order of the elements, and removing the `z-index` of them.

It continues using the `has-modal-open` class for backward compatibility (solution introduced in https://github.com/WordPress/gutenberg/pull/66093), but when the code is updated it's not used anymore (after the save).

## Testing Instructions





1. Before switching to this branch, add a Cover block to a post.
2. Inside the cover block, add a Navigation block.
3. After the previously created Cover block, add another Cover block.
4. Switch to this branch.
5. Visit the site in the frontend on small screens and also visit the editor. Open the menu, and make sure it's displayed over the whole content.
6. Edit something in the post, and save to update the deprecated version of the block.
7. Check that the frontend continues working properly.

## Screenshots or screencast 

https://github.com/user-attachments/assets/98bdfda3-4608-4869-90b1-887b09eaab89

Since I continued seeing the same result in my tests, I reused the same video from https://github.com/WordPress/gutenberg/pull/66093 just to illustrate the tests.
